doc/howtos/rdtraining/13_inheritance.rst

```diff
@@ -0,0 +1,244 @@
+and :meth:`~odoo.models.Model.unlink`.
+
+These methods (and any other method defined on :class:`~odoo.models.Model`) can be extended to add
+specific business logic::
+
+    from odoo import fields, models
+
+    class TestModel(models.Model):
+        _name = "test.model"
+        _description = "Test Model"
+
+        ...
+
+        @api.model
+        def create(self, vals):
+            # Do some business logic, modify vals...
+            ...
+            # Then call super to execute the parent method 
+            return super().create(vals)
+
+The decorator :func:`~odoo.api.model` is only necessary for the :meth:`~odoo.models.Model.create`
+method since the content of the recordset ``self`` is not relevant in the context of a creation.
+
+In Python 3, ``super()`` is equivalent to ``super(TestModel, self)``. The latter may be necessary
+in case you need to call the parent method with a modified recordset.
+
+.. danger::
+
+    - It is very important to **always** call ``super()`` to avoid breaking the flow. There are
+      only a few very specific cases where you don't want to do it.
+    - Pay attention to **always** return data consistent with the parent method. For example, if
+      the parent method returns a ``dict()``, your override must return a ``dict()``.
+
+.. exercise:: Add business logic to CRUD methods
+
+    - Prevent the deletion of a property if its state is not 'New' or 'Canceled'
+
+    Tip: override :meth:`~odoo.models.Model.unlink`, and be aware that ``self`` can be a recordset
+    with more than one record.
+
+    - At offer creation, set the property state to 'Offer Received' and raise an error if the user
+      tries to create an offer with a lower amount than an existing offer.
+
+    Tip: the ``property_id`` field is available in the ``vals``, but it is an ``int``. To
+    instanciate an ``estate.property`` object, use ``self.env[model_name].browse(value)``
+    (`example <https://github.com/odoo/odoo/blob/136e4f66cd5cafe7df450514937c7218c7216c93/addons/gamification/models/badge.py#L57>`__)
+
+Model Inheritance
+=================
+
+**Reference**: the documentation related to this topic can be found in
+:ref:`reference/orm/inheritance`.
+
+In our real estate module, we would like to display the list of properties linked to a salesperson,
+directly in the Settings / Users & Companies / Users form view. To do so, we need a field on the
+``res.users`` model, adapt the view to add it.
+
+Odoo provides two *inheritance* mechanisms to extend an existing model in a modular way.
+
+The first inheritance mechanism allows a module to modify the behavior of a model defined in
+another module:
+
+- add fields to a model,
+- override the definition of fields on a model,
+- add constraints to a model,
+- add methods to a model,
+- override existing methods on a model.
+
+The second inheritance mechanism (delegation) allows to link every record of a
+model to a record in a parent model, and provides transparent access to the
```

Comment:
```suggestion
The second inheritance mechanism (delegation) allows linking every record of a
model to a record in a parent model, and provides transparent access to the
```
Might deserve a warning or note that:
1. this delegation does *not* include methods
2. overriding fields in the "child" will not affect the parent, or apply any of the parent's behaviours relative to that field (e.g. constraints are not checked), it just "shadows" the delegated field
3. this is *not* prototypal inheritance, it's just a shortcut

In fact since related fields were already covered (I think?) it might be worth explaining that this is literally just convenience syntax to create a related field *for every field in the linked model*.